lib/internal/repl.js

```diff
@@ -55,15 +55,26 @@ function createRepl(env, opts, cb) {
   }
 
   const repl = REPL.start(opts);
-  if (opts.terminal && env.NODE_REPL_HISTORY !== '') {
+  if (opts.terminal) {
     return setupHistory(repl, env.NODE_REPL_HISTORY,
                         env.NODE_REPL_HISTORY_FILE, cb);
   }
+
   repl._historyPrev = _replHistoryMessage;
   cb(null, repl);
 }
 
 function setupHistory(repl, historyPath, oldHistoryPath, ready) {
+  // Empty string disables persistent history.
```

Comment:
I think you want to do

```javascript
if (typeof historyPath === 'string')
  historyPath = historyPath.trim();
```

Then drop the condition that is currently on line 74.